microprofile/metrics-smallrye/src/main/java/org/wildfly/extension/microprofile/metrics/MicroProfileMetricsSubsystemDefinition.java

```diff
@@ -73,19 +72,27 @@ public class MicroProfileMetricsSubsystemDefinition extends PersistentResourceDe
             .setAllowExpression(true)
             .build();
 
-    static final AttributeDefinition[] ATTRIBUTES = { SECURITY_ENABLED, EXPOSED_SUBSYSTEMS, PREFIX };
+    static final AttributeDefinition[] ATTRIBUTES = {SECURITY_ENABLED, EXPOSED_SUBSYSTEMS, PREFIX};
 
     protected MicroProfileMetricsSubsystemDefinition() {
         super(new SimpleResourceDefinition.Parameters(MicroProfileMetricsExtension.SUBSYSTEM_PATH,
                 MicroProfileMetricsExtension.getResourceDescriptionResolver(MicroProfileMetricsExtension.SUBSYSTEM_NAME))
-                .setAddHandler(MicroProfileMetricsSubsystemAdd.INSTANCE)
+                .setAddHandler(new ModelOnlyAddStepHandler(ATTRIBUTES))
                 .setRemoveHandler(new ReloadRequiredRemoveStepHandler())
-                .setCapabilities(MICROPROFILE_METRIC_HTTP_CONTEXT_CAPABILITY, MICROPROFILE_METRICS_HTTP_SECURITY_CAPABILITY,
-                        MICROPROFILE_METRICS_SCAN));
+                .setCapabilities(MICROPROFILE_METRICS_HTTP_SECURITY_CAPABILITY, MICROPROFILE_METRICS_SCAN));
+    }
+
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+        for (AttributeDefinition att : ATTRIBUTES) {
+            resourceRegistration.registerReadWriteAttribute(att, null,
+                    new ReloadRequiredWriteAttributeHandler(ATTRIBUTES));
+        }
```

Comment:
@jasondlee  FYI -- if you pass ATTRIBUTES as a param to the superclass constructor, you don't need to override this method.

The effect is the same, just a bit less code, so I'm not asking you to change this unless you change this branch for a different reason.

Reply:
Noted. Thanks.